opossum-tool/OpossumUI
# feat: fine-tune autocomplete suggestions

### Summary of changes

1. ensure that the autocomplete suggestion "+" button overwrites all previously entered data so that there cannot be confusion about what was and what wasn't replaced by the user's action
2. make autocomplete suggestion "+" button more visually distinct from the primary suggestion action of just adding the single attribute so that it becomes clearer that there really are two separate actions on the suggestion card
![image](https://github.com/opossum-tool/OpossumUI/assets/46091730/a5b03ee6-9e7f-4438-bdbb-eb6a730ec83e)
3. ensure click on entire autocomplete input field opens option popper and not just a click on the underlying HTML input element
4. catch unhandled errors produced by the PackageURL constructor and instead display "INVALID PURL"
5. do not set a default version when user adopts a suggestion via autocomplete "+" button because the default (latest) version will almost always be the wrong one

### Context and reason for change

Finalizing #2333.

### How can the changes be tested

See summary of changes.

For 3: click on the margin area of the input element (highlighted in green here): 
![image](https://github.com/opossum-tool/OpossumUI/assets/46091730/75bc6f9c-49cc-4353-aa85-a6c8ffd5bb94)

For 4: chose "pub" as package type and type a package name which includes an extended symbol. See [here](https://github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst) for details.
![image](https://github.com/opossum-tool/OpossumUI/assets/46091730/d3acc5db-70b8-40ae-a7b2-5661a46ac040)
